test/libsolidity/SemanticTest.cpp

```diff
@@ -204,6 +233,20 @@ void SemanticTest::printUpdatedExpectations(ostream& _stream, string const&) con
 		_stream << test.format("", true, false) << endl;
 }
 
+void SemanticTest::printUpdatedSettings(ostream& _stream, string const& _linePrefix)
+{
+	auto& settings = m_reader.settings();
+	if (settings.empty() && !m_compileViaYulCanBeSet)
+		return;
+
+	_stream << _linePrefix << "// ====" << endl;
+	if (m_compileViaYulCanBeSet)
```

Comment:
Is this ever true if we run `compileViaYul: only`?

Reply:
It is never true in that case. It is true only if it is missing compileViaYul, or if it is explicitly set to false. Anyway if it makes confusion, I can try making it more readable

Reply:
it is set to false when we start executing test and it is set to true with
```
if (success && !m_runWithYul && compileViaYul)
{
	m_compileViaYulCanBeSet = true;
	...
}
```

So, it will be set only if test succeded with compileViaYul and m_runWithYul wasn't selected, so should be fine